### DISCUSSION TITLE
Problem with test FundMe.sol // [FAIL: assertion failed: 6 != 4] testPriceFeedVersionIsAccurate() (gas: 18905)

### DISCUSSION THREAD

**Discussion by pollersan** [Cyfrin/foundry-full-course-cu#2597] [Q&A] [answered]
Hello!

I am having a problem using: “forge test --fork-url $MAINNET_RPC_URL” I get this error:

```
Ran 3 tests for test/FundMeTest.t.sol:FundMeTest
[PASS] testMinimumDollarIsFive() (gas: 8323)
[PASS] testOwnerIsMsgSender() (gas: 10644)
[FAIL: assertion failed: 6 != 4] testPriceFeedVersionIsAccurate() (gas: 18905)
Suite result: FAILED. 2 passed; 1 failed; 0 skipped; finished in 2.17s (510.39ms CPU time)

Ran 1 test suite in 2.57s (2.17s CPU time): 2 tests passed, 1 failed, 0 skipped (3 total tests)

Failing tests:
Encountered 1 failing test in test/FundMeTest.t.sol:FundMeTest
[FAIL: assertion failed: 6 != 4] testPriceFeedVersionIsAccurate() (gas: 18905)

Encountered a total of 1 failing tests, 2 tests succeeded

```
Here is my code:

FundMeTest.t.sol

```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../src/FundMe.sol";
import {DeployFundMe} from "../script/DeployFundMe.s.sol";

contract FundMeTest is Test {
    FundMe fundMe;

    function setUp() external {
        // us -> FundMeTest -> FundMe
        // fundMe = new FundMe(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        DeployFundMe deployFundMe = new DeployFundMe();
        fundMe = deployFundMe.run();
    }

    function testMinimumDollarIsFive() public {
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }

    function testOwnerIsMsgSender() public {
        assertEq(fundMe.i_owner(), msg.sender);
    }

    function testPriceFeedVersionIsAccurate() public {
        uint256 version = fundMe.getVersion();
        assertEq(version, 4);
    }
}

```

I've followed all the steps in the course, but I'm hitting this error. I've spent over 30 minutes trying to figure it out on my own, and it seems like the issue is related to the ETH Mainnet RPC using version 6. 

Thank you for you time.

**Comment by EngrPips** [comment]
Hello @pollerpan, The test is failing because the version of the ETH/USD price feed contract on Ethereum is actually `6` and not `4`

**Reply by pollersan** [reply]
Hello @EngrPips ,

I'm using the same code from the course, but I'm running into this issue, and he isn’t. Could it be that I did something wrong, or has the version changed recently? I want to make sure I'm following the correct steps.

**Reply by EngrPips** [reply]
He filmed that section a few months before now I believe, and things constantly change in the programming space. So I will assume things change and your code is fine as far as I know. you can change the `4` in your test to `6` and the test should pass.

**Reply by pollersan** [reply]
What I did was modify that part of the code so the RPC works depending on the network I'm using, which helps me avoid the error. I really appreciate your response; I just wanted to make sure I wasn't missing anything.

Code updated for me:

```solidity
    function testPriceFeedVersionIsAccurate() public {
        uint256 version = fundMe.getVersion();
           uint256 expectedVersion; // Cambiar a uint256

    if (block.chainid == 11155111) { // Sepolia
        expectedVersion = 4; // Adjust version for Sepolia
    } else if (block.chainid == 1) { // Mainnet
        expectedVersion = 6; // Adjust version for Mainnet
    } else if (block.chainid == 31337) { // Anvil
        expectedVersion = 4; // Adjust version for Anvil 
    } else {
        revert("Unsupported chain"); // Unsupported chain
    }

    assertEq(version, expectedVersion);
    }

    function testCurrentChainId() public {
        console.log("Current chain ID:", block.chainid);
    }
}

```

**Reply by EngrPips** [reply, marked as answer]
> What I did was modify that part of the code so the RPC works depending on the network I'm using, which helps me avoid the error. I really appreciate your response; I just wanted to make sure I wasn't missing anything.
> 
> Code updated for me:
> 
> ```solidity
>     function testPriceFeedVersionIsAccurate() public {
>         uint256 version = fundMe.getVersion();
>            uint256 expectedVersion; // Cambiar a uint256
> 
>     if (block.chainid == 11155111) { // Sepolia
>         expectedVersion = 4; // Adjust version for Sepolia
>     } else if (block.chainid == 1) { // Mainnet
>         expectedVersion = 6; // Adjust version for Mainnet
>     } else if (block.chainid == 31337) { // Anvil
>         expectedVersion = 4; // Adjust version for Anvil 
>     } else {
>         revert("Unsupported chain"); // Unsupported chain
>     }
> 
>     assertEq(version, expectedVersion);
>     }
> 
>     function testCurrentChainId() public {
>         console.log("Current chain ID:", block.chainid);
>     }
> }
> ```

You are a giga chad sir

**Reply by liumOazed** [reply]
Hi @pollerpan awesome work! Just wondering how did you get anvil chainid? I was searching it on chainlist.org :)

**Reply by EngrPips** [reply]
You get the chainid by running `Anvil` on your terminal, as shown in the image below. 

<img width="1440" alt="Screenshot 2024-10-06 at 17 07 43" src="https://github.com/user-attachments/assets/6339dd56-57d1-4b7f-99dc-713cf8b4a741">

**Reply by liumOazed** [reply]
Awesome @EngrPips :)

**Reply by EngrPips** [reply]
You are welcome, Friend.

**Reply by 0xsl0th** [reply]
Thanks @EngrPips !!

**Reply by EngrPips** [reply]
Anytime, Friend. Keep crushing it.
